app/controllers/api/v2/vcs_clone_controller.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class VcsCloneController < ::Api::V2::BaseController
+      include ::ForemanAnsible::ProxyAPI
+
+      rescue_from ActionController::ParameterMissing do |e|
+        render json: { 'error' => e.message }, status: :bad_request
+      end
+
+      rescue_from Foreman::Exception do |_e|
+        head :internal_server_error
+      end
+
+      skip_before_action :verify_authenticity_token
+
+      before_action :set_proxy_api
+
+      api :GET, '/smart_proxies/:proxy_name/ansible/vcs_clone/repo_information',
+          N_('Queries metadata about the repo')
+      param :proxy_name, Array, N_('Name of the SmartProxy'), :required => true
+      param :vcs_url, String, N_('Url of the repo'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled / invalid repo-info')
+      def repo_information
+        vcs_url = params.require(:vcs_url)
+        render json: @proxy_api.repo_information(vcs_url)
+      end
+
+      api :GET, '/smart_proxies/:proxy_name/ansible/vcs_clone/roles',
+          N_('Returns an array of roles installed on the provided proxy')
+      formats ['json']
+      param :proxy_name, Array, N_('Name of the SmartProxy'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled')
+      def installed_roles
+        render json: @proxy_api.list_installed
+      end
+
+      api :POST, '/smart_proxies/:proxy_name/ansible/vcs_clone/roles',
+          N_('Launches a task to install the provided role')
+      formats ['json']
+      param :repo_info, Hash, :desc => N_('Dictionary containing info about the role to be installed') do
+        param :vcs_url, String, :desc => N_('Url of the repo'), :required => true
+        param :name, String, :desc => N_('Name of the repo'), :required => true
+        param :ref, String, :desc => N_('Branch / Tag / Commit reference'), :required => true
+      end
+      param :smart_proxy, Array, N_('SmartProxy the role should get installed to')
+      error 400, :desc => N_('Parameter unfulfilled')
+      def install_role
+        payload = verify_task_parameters(params)
+        start_vcs_task(payload, :install)
+      end
+
+      api :PUT, '/smart_proxies/:proxy_name/ansible/vcs_clone/roles',
+          N_('Launches a task to update the provided role')
+      formats ['json']
+      param :repo_info, Hash, :desc => N_('Dictionary containing info about the role to be installed') do
```

Comment:
Semantically I'd expect a `PUT` on a collection to ensure that the provided list of items is the exact set of items present. Or put in other words, `PUT` on `.../roles` with `my_role` would ensure that only `my_role` is present and all other roles are removed.

If the name is immutable (and I think it's safe to assume so) you could make it part of the URL and make it `PUT .../roles/:name` to update things (like VCS URL and ref). Just like you did with `DELETE`.

Reply:
I added `role_name` as a parameter to the URL, which does indeed make more sense now!